src/esphome/sensor/binary_sensor_map.h

```diff
@@ -0,0 +1,81 @@
+#ifndef ESPHOME_SENSOR_BINARY_SESNOR_MAP_H
+#define ESPHOME_SENSOR_BINARY_SESNOR_MAP_H
+
+#include "esphome/defines.h"
+
+#ifdef USE_BINARY_SENSOR_MAP
+
+#include "esphome/component.h"
+#include "esphome/sensor/sensor.h"
+#include "esphome/binary_sensor/binary_sensor.h"
+
+ESPHOME_NAMESPACE_BEGIN
+
+namespace sensor {
+
+enum {
+  BINARY_SENSOR_MAP_TYPE_GROUP = 0x00,
+  BINARY_SENSOR_MAP_TYPE_SLIDER = 0x01,
+  BINARY_SENSOR_MAP_TYPE_WHEEL = 0x02,
+};
+
+/** helper class to combine the binary_sensor and its float value to one objects. **/
+class BinarySensorMapChannel {
```

Comment:
If this is only going to be a data class, and has no logic itself then make it a `struct`. Otherwise make the attributes protected and write getters.